packages/front-end/components/Radix/Button.tsx

```diff
@@ -0,0 +1,99 @@
+import { Button as RadixButton, ButtonProps, Text } from "@radix-ui/themes";
+import { CSSProperties, ReactNode } from "react";
+import { Responsive } from "@radix-ui/themes/dist/cjs/props";
+import Link from "next/link";
+import ConditionalWrapper from "@/components/ConditionalWrapper";
+
+type Overwrite<T, NewT> = Omit<T, keyof NewT> & NewT;
+
+export type Color = "primary" | "danger";
+export type Variant = "solid" | "soft" | "outline" | "ghost";
+export type Size = "xs" | "sm" | "md";
+
+export type Props = Overwrite<
+  ButtonProps,
+  {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onClick?: (e: any) => void;
+    href?: string;
```

Comment:
Wonder if we should split these into a `ButtonLink` component (with required `href` prop) and a `Button` component (with required `onClick` prop).  Possibly even a 3rd one for `SubmitButton` (with neither `href` or `onClick`).  They can share 99% of the code under the hood, but splitting them would help avoid mistakes.

Also, for the `onClick` prop, do we need to pass the event in as an argument?  In the other Button prop, it automatically calls `e.preventDefault()`, which I think we should do as well.  Then, there's no need to have the event.

Reply:
Re: ButtonLink and SubmitButton, I'm on the fence about this. As a dev, I think I would expect a button to behave at least largely like a native HTML button, and wouldn't think that I'd need to reference 3 different things. We'll likely have some combinatory issues as well if we decide to do <TextButton> or any other <StyledButton> supersets.

onClick: Yeah I guess we almost never need the underlying event. It does make me think though that perhaps we need to fully implement the async + auto-loading states, like in the old Button.

Reply:
Added the async loading stuff. lmk what we want to do about links and other button variants. I'm ok splitting them if you prefer.

Reply:
Decided to implement a LinkButton. Thinking we hold off on SubmitButton for now because its so similar to Button.